modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -160,6 +169,39 @@ def get_document_summaries(veis_token, btsss_token, claim_id)
       documents
     end
 
+    # Disabled method length for now due to error handling adding extra lines
+    # Once we refactor all Travel Pay services for better error handling we can adjust the length here
+    def loop_and_paginate_claims(params, veis_token, btsss_token) # rubocop:disable Metrics/MethodLength
+      page_number = params['page_number'] || 1
+      all_claims = []
+      total_record_count = 0
+
+      client_params = params.merge!({ page_number: })
+      faraday_response = client.get_claims_by_date(veis_token, btsss_token, client_params)
+      total_record_count = faraday_response.body['totalRecordCount']
+      all_claims.concat(faraday_response.body['data'].deep_dup)
+
+      while all_claims.length < total_record_count
+        page_number += 1
+
+        client_params[:page_number] = page_number
+        faraday_response = client.get_claims_by_date(veis_token, btsss_token, client_params)
+
+        all_claims.concat(faraday_response.body['data'])
+      end
+
+      { data: all_claims, total_record_count:, page_number:, status: 200 }
+    rescue => e
```

Comment:
Can we be more specific about the type of error we're rescuing here? I'm thinking Faraday::Error (covers all client and server error types)

Reply:
Oh why's that?